lib/posix/options/mlock.c

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2024, Tenstorrent AI ULC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <stddef.h>
+#include <sys/types.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/posix/sys/mman.h>
```

Comment:
nit: we probably have more headers here than we need

Reply:
Fixed!